### PR TITLE
Biogenerator Powercreep + Improvements

### PR DESCRIPTION
**Pull request by BurgerLUA**
The biogenerator seemed pretty useless on top of a generally boring job. There wasn't really anything interesting you could make with the default biogenerator recipes so I went on a quest to add more useful recipes that make sense. I looked at the code and it was pretty shit (everything was hardcoded, and there was also an exploit involving monkeys) so I added a better framework for adding biogenerator recipes, and added multiple item making.

Previously, you used to be able to make:
- 10u Milk at a time that went straight into the bottle (wew)
- Slabs of Meat, which is the only thing useful assuming security didn't powergame on the chef's behalf and get the cow from maint.
- E-Z Nutrient for growing more plants.
- Left 4 Zed which I think doesn't even work.
- Robust Harvest for growing more plants.
- A Wallet in case you want to save your one loadout slot.
- Botanical Gloves, which really only benefits xenobotany.
- A utility belt which is honestly the most useful item in there that doesn't involve your job.
- A satchel for carrying all the meat you made.
- A cash bag that's there to taunt you by daydreaming about vault heists.

With this update, now you get to make:
- Biomeat which is just regular meat but with an appropriate appearance.
- 50u Milk Cartons instead of 10u milk every activation.
- Soylent Viridans!
- Vitamin Pills that contain nutrients and random juice flavoring.
- Food Rations for Loyalist/Rev Rounds.
- Universal Enzyme (in case it ever gets stolen or misplaced by the chef)
- Duct tape for antagonism and general usefulness. (Good for that special spot in maint that botany has access to.)
- Soap made from plants!
- AMAZING TOWELS!
- Animal Hides, Leather, Cloth, and Cardboard Sheets for construction!
- Wax for making... candles...
- WALKING MUSHROOM PETS.

But if you're an antag who's balsy and use the emag on the biogenerator, you get to make:
- Bruise packs and ointment that heal brute and burn damage.
- Human Hide for construction.
- Soylent Green for memes.
- Syndie Soap for memes.
- Buckler Shields.
- k i l l e r   p i n e t r e e s

Screenshots:

https://i.imgur.com/sCD2BIn.png
https://i.imgur.com/5ZPfGr1.png

### CODE REVIEW

File: html/changelogs/burgerbb-biogenerator.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+author: BurgerBB
+delete-after: True
+changes: 
+  - tweak: "Improved code and UI."
```
Comment (skull132) [inline html/changelogs/burgerbb-biogenerator.yml]: This, and the report below it should be rolled into one. And given context. Right now it's unclear as to _what_ had it's code and UI improved.

File: html/changelogs/burgerbb-biogenerator.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  - tweak: "Changed and added some new sounds."
+  - tweak: "When creating milk, cartons of milk are spawned instead of being put into the bottle."
+  - rscadd: "Added the ability to make multiple items of the same type in the Biogenerator for all recipies."
+  - rscadd: "Added Biomeat; a differently colored variant of regular meat with different flavoring, but still has the functionality of meat. Still unsafe for Skrell consumption."
+  - rscadd: "Added Vitamin Pills that contain 25 nutrients and 1 random flavoring of juice."
+  - rscadd: "Added additional biogenerator recipies."
+  - rscadd: "Added the ability to emag the biogenerator. Emagged Biogenerators unlocked hidden recipies."
```
Comment (skull132) [inline html/changelogs/burgerbb-biogenerator.yml]: Roll this into the first line as well.

File: html/changelogs/burgerbb-biogenerator.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+author: BurgerBB
+delete-after: True
+changes: 
+  - tweak: "Improved code and UI."
+  - tweak: "Changed and added some new sounds."
+  - tweak: "When creating milk, cartons of milk are spawned instead of being put into the bottle."
+  - rscadd: "Added the ability to make multiple items of the same type in the Biogenerator for all recipies."
+  - rscadd: "Added Biomeat; a differently colored variant of regular meat with different flavoring, but still has the functionality of meat. Still unsafe for Skrell consumption."
+  - rscadd: "Added Vitamin Pills that contain 25 nutrients and 1 random flavoring of juice."
+  - rscadd: "Added additional biogenerator recipies."
```
Comment (skull132) [inline html/changelogs/burgerbb-biogenerator.yml]: Roll this and the bottom most one into one.

**Comment by Lohikar** [comment]
Maybe true NUI or fake NUI might be better than tables.

**Comment by BurgerLUA** [comment]
Formatting looks like this now:
https://i.imgur.com/1dR72CV.png
I cannot stress this enough, but I absolutely cannot code in NUI. I spent an hour on it yesterday and it was just too stressful to deal with. Perhaps one day someone else can convert it to NUI.

**Pull request merged**